### PR TITLE
Jakub improve explorer performance 184

### LINKED ISSUE #184: Improve map explorer performance
Display a maximum number of images that display per page when viewing images on a camera station. Currently if one camera station has 1000s of images I think the app tries to load them all into one page, which makes it very slow

### PR DESCRIPTION
**Pull request by jchojna**
Closes #184 

### Description
- use react-window to virtualize huge list of observations, in order to improve loading performance
- add new typescript's typings
- fade out scrolled card just a bit
- add loader instead of observation photo, when the card is being loaded

### How to test
- run the app and test observations panel manually
- check why installed types for react-window-infinite-loader are not detected automatically

### CODE REVIEW

**Review by kamilzyla — CHANGES_REQUESTED** [review]
The core of the solution works :+1: :tada: Great job on researching how to approach this! Let's take some time to simplify the mechanism a bit.

In addition to my comments you might want to take a look at [React Virtualization - react-window vs react-virtuoso](https://dev.to/sanamumtaz/react-virtualization-react-window-vs-react-virtuoso-8g). This article contains a nice summary of various packages which can be used for virtualization. Consider whether we could use `react-virtuoso`: it seems that it would do the job easily with just one dependency. Don't worry about it if you figure it would take too much work.

File: app/components/ObservationsInspector.scss [hunk 1]
```diff
@@ -1,27 +1,113 @@
-.card {
-  margin-top: 10px;
+@use '../styles/variables' as var;
 
-  .heading {
-    margin-top: 0;
+.drawer {
+  width: 750px !important; // overriding specificity would be too complicated
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/components/ObservationsInspector.scss:4]: ahhh.. good point! I missed that prop. Fixed!

File: app/components/ObservationsInspector.tsx [hunk 2]
```diff
@@ -114,6 +125,61 @@ type ObservationsInspectorProps = {
...
+    observations: Observation[];
+    predictionOverrides: PredictionOverridesMap;
+    onPredictionOverride: PredictionOverrideHandler;
+  };
+};
+
+const ObservationRow = ({ index, isScrolling, style, data }: ObservationRowProps) => {
+  const { observations, predictionOverrides, onPredictionOverride } = data;
+  const observation = observations[index];
+  if (observation === undefined) return false;
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/components/ObservationsInspector.tsx:161]: Frankly speaking, this line is more a leftover now, I just realized. I needed to handle undefined case for observation at some point, and forgot to refactor it. Right now it seems redundant. I removed that line

File: app/components/ObservationsInspector.scss [hunk 3]
```diff
@@ -1,27 +1,113 @@
...
+
+  .drawer-body {
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/components/ObservationsInspector.scss:4]: heh.. corrected :)

File: app/styles/_variables.scss [hunk 4]
```diff
@@ -21,3 +21,4 @@ $radius-card: 10px;
 // Transitions
 $t-marker: 0.2s;
 $t-marker-animation: 1.5s;
+$t-observation-card: 0.5s;
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/styles/_variables.scss:24]: Agree. Fixed!

File: app/components/ObservationsInspector.tsx [hunk 5]
```diff
@@ -125,22 +191,44 @@ export default function ObservationsInspector(props: ObservationsInspectorProps)
...
-            <ObservationCard
-              key={observation.location}
-              observation={observation}
-              predictionOverride={predictionOverrides[observation.location]}
-              onPredictionOverride={onPredictionOverride}
-            />
-          ))}
+      <div className={`${Classes.DRAWER_BODY} ${styles['drawer-body']}`}>
+        <div className={`${Classes.DIALOG_BODY} ${styles['dialog-body']}`}>
+          <InfiniteLoader
```
[the remark this answers is not in the document]
Comment (jchojna) [reply app/components/ObservationsInspector.tsx:204]: :open_mouth:

File: package.json [hunk 6]
```diff
@@ -310,6 +314,10 @@
     "react-i18next": "^11.17.4",
     "react-router-dom": "^5.1.2",
     "react-select": "^5.4.0",
+    "react-spinners": "^0.13.4",
```
[the remark this answers is not in the document]
Comment (jchojna) [reply package.json:317]: Done

File: app/components/ObservationsInspector.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (jchojna) [reply app/components/ObservationsInspector.tsx:204]: I didn't realize that having just `FixedSizeList` will do the trick here. Fixed!

**Review by kamilzyla — APPROVED** [review]
LGTM, works well! Great job :grin: :100:

**Pull request merged**